java/org/openobservatory/measurement_kit/nettests/OoniTestBase.java

```diff
@@ -0,0 +1,74 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+package org.openobservatory.measurement_kit.nettests;
+
+import org.openobservatory.measurement_kit.android.DnsUtils;
+import org.openobservatory.measurement_kit.common.LogCallback;
+import org.openobservatory.measurement_kit.swig.OoniTestWrapper;
+
+public class OoniTestBase {
+    private OoniTestWrapper wrapper = null;
+
+    public OoniTestBase(String test_name) {
+        wrapper = new OoniTestWrapper(test_name);
+
+        // Rationale: start with reasonable DNS configuration and then the user is
+        // free to override it using calling again `set_options` if need be.
+        set_options("dns/nameserver", DnsUtils.get_device_dns());
```

Comment:
Should I remove these lines from the android app?

```
        // The app now tries to get DNS from the device. Upon fail, it uses
        // Google DNS resolvers
        String nameserver_ = "8.8.8.8";
        ArrayList<String> nameservers = getDNS();
        if (!nameservers.isEmpty()) {
            for (String s : getDNS()) {
                nameserver_ = s;
                Log.v(TAG, "Adding nameserver: " + s);
                break;
            }
        }
        final String nameserver = nameserver_+":53";
```

`w.set_options("dns/nameserver", nameserver);`